common/src/main/java/io/netty/util/ResourceLeakTracker.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util;
+
+public interface ResourceLeakTracker<T>  {
+
+    /**
+     * Records the caller's current stack trace so that the {@link ResourceLeakDetector} can tell where the leaked
+     * resource was accessed lastly. This method is a shortcut to {@link #record(Object) record(null)}.
+     */
+    void record();
+
+    /**
+     * Records the caller's current stack trace and the specified additional arbitrary information
+     * so that the {@link ResourceLeakDetector} can tell where the leaked resource was accessed lastly.
+     */
+    void record(Object hint);
+
+    /**
+     * Close the leak so that {@link ResourceLeakTracker} does not warn about leaked resources.
```

Comment:
Maybe consider rewording or adding the following:

After this method is called a leak associated with this `ResourceLeakTracker` should not be reported.